src/components/Square.js

```diff
@@ -1,24 +1,32 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-import './Square.css'
+import './Square.css';
 
 const Square = (props) => {
   // For Wave 1 enable this 
   //  Component to alert a parent 
   //  component when it's clicked on.
 
-  return <button
-    className="square"
-  >
-    {props.value}
-  </button>
-}
+  const onSquareClick = () => {
+    const markedSquare = {
+      value: props.playerTurn,
+      id: props.id
+    };
+    props.onClickCallback(markedSquare);
+  };
+
+  return <button disabled={props.winnerState !== null} 
+  onClick={onSquareClick} className='square'>{props.value}</button>;
```

Comment:
Rather than disabling the buttons (based on whether the game is over) we can add logic to the click handler back in App to prevent processing any clicks after the game is over. That would allow us to handle the required functionality without needing to introduce additional props (which causes problems for the tests).

In your own application, your approach would be perfectly reasonable and is up to you.